### ISSUE TITLE
Problem connecting to the network after updating the Arduino libraries.

### ISSUE THREAD

**Issue by SteveRMann** [arendst/Sonoff-Tasmota#3684]
Make sure these boxes are checked [x] before submitting your issue - Thank you!

- [x ] Searched the problem in issues (https://github.com/arendst/Sonoff-Tasmota/issues)
- [x ] Searched the problem in the wiki (https://github.com/arendst/Sonoff-Tasmota/wiki/Troubleshooting)
- [x ] Searched the problem in the forum (https://groups.google.com/d/forum/sonoffusers)
- [x ] Searched the problem in the chat (https://discord.gg/Ks2Kzd4)
- [x ] Hardware used : 
- [x ] Provide the output of command ``status 0`` :
```
status 0
00:01:20 CMD: status 0status 0status 0
00:01:20 RSL: stat/rum/STATUS = {"Status":{"Module":1,"FriendlyName":["Xyzzy Labs"],"Topic":"rum","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":6,"SaveData":1,"SaveState":1,"ButtonRetain":0,"PowerRetain":0}}
00:01:20 RSL: stat/rum/STATUS1 = {"StatusPRM":{"Baudrate":115200,"GroupTopic":"sonoffs","OtaUrl":"http://sonoff.maddox.co.uk/tasmota/sonoff.bin","RestartReason":"Power on","Uptime":"0T00:01:20","StartupUTC":"","Sleep":0,"BootCount":28,"SaveCount":52,"SaveAddress":"FA000"}}
00:01:20 RSL: stat/rum/STATUS2 = {"StatusFWR":{"Version":"6.1.1.14","BuildDateTime":"2018-09-01T18:14:36","Boot":6,"Core":"2_4_2","SDK":"2.2.1(cfd48f3)"}}
00:01:20 RSL: stat/rum/STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["Kaywinnet",""],"TelePeriod":300,"SetOption":["00008009","55818000","00000001"]}}
00:01:20 RSL: stat/rum/STATUS4 = {"StatusMEM":{"ProgramSize":477,"Free":524,"Heap":26,"ProgramFlashSize":1024,"FlashSize":1024,"FlashMode":3,"Features":["00000809","0FC8E790","00000000","02B400CE","00000000"]}}
00:01:20 RSL: stat/rum/STATUS5 = {"StatusNET":{"Hostname":"qwerty","IPAddress":"0.0.0.0","Gateway":"192.168.1.1","Subnetmask":"255.255.255.0","DNSServer":"192.168.1.1","Mac":"60:01:94:69:2D:B5","Webserver":2,"WifiConfig":5}}
00:01:20 RSL: stat/rum/STATUS6 = {"StatusMQT":{"MqttHost":"192.168.1.124","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_692DB5","MqttUser":"DVES_USER","MqttType":1,"MAX_PACKET_SIZE":1000,"KEEPALIVE":15}}
00:01:20 RSL: stat/rum/STATUS7 = {"StatusTIM":{"UTC":"Thu Jan 01 00:01:20 1970","Local":"Thu Jan 01 00:01:20 1970","StartDST":"Thu Jan 01 00:00:00 1970","EndDST":"Thu Jan 01 00:00:00 1970","Timezone":99,"Sunrise":"12:15","Sunset":"21:22"}}
00:01:20 RSL: stat/rum/STATUS10 = {"StatusSNS":{"Time":"1970-01-01T00:01:20","Switch1":"ON"}}
00:01:20 RSL: stat/rum/STATUS11 = {"StatusSTS":{"Time":"1970-01-01T00:01:20","Uptime":"0T00:01:20","Vcc":3.392,"POWER":"OFF","Wifi":{"AP":1,"SSId":"Kaywinnet","RSSI":100,"APMac":"44:D1:FA:1E:94:3D"}}}
00:01:26 WIF: Connect failed as AP cannot be reached
00:01:26 WIF: Connecting to AP1 Kaywinnet in mode 11N as qwerty...
```

I have a problem with the Sonoff connecting to the network that just developed after updating the Arduino libraries.

I started with five Sonoff Basic switches and was using the Arduino IDE to flash them with Tasmota.  (Each with a different project name).    The first three flashed and connected to the network just fine.  When the flash finished the LED blinked a few seconds then stopped, at which time I opened the web console to finish the configuration.  I took a break and when I returned, the IDE was prompting me to update the libraries- which I did.  And that's when the problem began.  Now, the sketch compiles OK and flashes OK- but the Sonoff just won't connect to the network.

I used the serial interface and Termite (a simple tiny terminal program) and could use the console commands - Tasmota was running, the AP name and password were correct, but it just won't connect to the net.

I downloaded a fresh copy of the Tasmota code in case I accidentally changed something, but no difference.

I also compiled a sketch that I had been using on the Sonoff's- it compiled and flashed OK and connected to the net as always.

I downloaded the latest sonoff.bin file and flashed it with ESPeasy, and everything works fine. It connects to the AP and I can change the parameters in the console.

But, when I flash from the IDE, the Sonoff just won't connect to the net.

**Comment by ascillato2** [comment]
Which Arduino IDE version?

I can not reproduce your issue neither Arduino nor Platformio. Works in both platforms, so the problem is not in Tasmota. It is in your IDE.

I have just tested on last platformio and arduino 1.8.2 and works

Have you tried to uninstall and install again?

**Comment by ascillato2** [comment]
Please, also try to erase all flash, and then flash last release.

**Comment by SteveRMann** [comment]
Thanks for the reply and tips.

Using a second PC.
I Installed the Arduino IDE (V 1.8.6) and downloaded a new copy of the Tasmota sources.  Not sure how to read the version, but sonoff_version.h contains: #define VERSION   0x0601010E.
Compiled and flashed Tasmota to the Sonoff.
No change, (WIF: Connect failed as AP cannot be reached)
I tried flashing a dummy sketch, just the void setup() and void main(), then flashed Tasmota to the Sonoff..
No change.
I also compiled and flashed the eeprom clear sketch (in examples), then flashed Tasmota to the Sonoff.
No change
I also tried a new Sonoff that has never been flashed with Tasmota.
No change
I also set a static IP, thinking possible DHCP problem.
No change
I also rebooted my Router as a "Hail Mary".  I don't like restarting my router as I have a lot of IP devices that need to reconnect, so it's a last thing to try.
No change.

I installed PlatformIO, and after several hours I was finally able to get a blink program to compile and flash.  I don't think I am ready to try importing Tasmota into PlatformIO since I have no clue what I am doing in PlatformIO and there's no "PlatformIO for Dummies" that I can find.   But that's another issue.

As I said in another post, I can flash the sonoff.bin image using ESPEasy and finish my project using serial I/O.  But I would like to figure out why when flashed via the Arduino IDE, the Sonoff can't find the net.  Especially when it worked on three previous Sonoff switches.

**Comment by Frogmore42** [comment]
You have it configured to use core 2.4.2.  While I am using core 2.4.2 and having success with it.  I did have issues getting it set up.  I believe the bin is compiled with core 2.3.0.  I would suggest that you either use the pre-compiled bin or you will have to invest the time necessary to become proficient in using Arduino or PlatformIO.  Arduino is really not designed for something as complicated as Tasmota, in my opinion.  I am pretty sure there are tutorials on using PlatformIO.  Many people using Tasmota have figured it out, so I am sure you can too.  Or you can read the wiki instructions for Arduino and selecting the core version if you prefer.  While PlatformIO can be very complex, you don't have to understand or use most of what it can do to just compile and download.  But, I do understand how frustrating it can be.  I am having that experience right now with OpenHAB.  PlatformIO is ten times easier to understand and better documented, but then it is a professional product.

**Comment by Frogmore42** [comment]
Oh and you need to try using core 2.3.0, which is why you need to be proficient in your choice of environment.  2.3.0 is likely to work better in your environment.  With 2.4.2, I had to COMPLETELY erase the flash, not the eeprom.  Esptool has the ability to do a complete erase before flashing.

**Comment by Frogmore42** [comment]
PlatformIO on VSCode has a very hand button for that, assuming you have downloaded a copy of the code.

VSCode has very handy integration with Git, which you can use if you clone the repo.  You need to have at least a basic understanding of Git to do this.

**Comment by SteveRMann** [comment]
As I said, I do not believe there's a problem with Tasmota code, but probably a library in my IDE.  For that reason I will close this issue to keep the Git clean.  

What's puzzling is that I can duplicate the problem of connecting to the net on a different PC.  (On PC2 I installed the Arduino IDE - PC2 had never seen Arduino before this.  I then downloaded Tasmota from the Git. Then I compiled and flashed Tasmota.  No errors anywhere along the process, but still, the Sonoff just would not connect to the AP.)

The only reason that I initially suspected Tasmota was because all of my other work on the IDE still compiles, flashes and connects to the same AP without problem.

My next step is to get more familiar with PlatformIO.
